src/Functions/splitByRegexp.cpp

```diff
@@ -79,7 +81,7 @@ class SplitByRegexpImpl
     /// Get the next token, if any, or return false.
     bool get(Pos & token_begin, Pos & token_end)
     {
-        if (!re)
+        if (!re) [[unlikely]]
```

Comment:
Tbh, I am skeptical about all `[[likely]]` and `[[unlikely]]` in this PR. Do they make a measurable difference?

Reply:
I'll measure it.

Reply:
@rschu1ze you are right. They doesn't make sense. 

splitByChar
```
with 'Many years later as he faced the firing squad, Colonel Aureliano Buendia was to remember that distant afternoon when his father took him to discover ice.' as s select splitByChar(' ', materialize(s)) as w from numbers(1000000) format Null;

With [[likely]] and [[unlikely]]: 
localhost:9001, queries: 10, QPS: 1.433, RPS: 1432671.078, MiB/s: 10.930, result RPS: 0.000, result MiB/s: 0.000.

Without [[likely]] and [[unlikely]]: 
localhost:9001, queries: 10, QPS: 1.610, RPS: 1609738.252, MiB/s: 12.281, result RPS: 0.000, result MiB/s: 0.000.
``` 


splitByRegexp (do not fallback)
```
with 'Many years later as he faced the firing squad, Colonel Aureliano Buendia was to remember that distant afternoon when his father took him to discover ice.' as s select splitByRegexp('\s+', materialize(s)) as w from numbers(100000) format Null;

With [[likely]] and [[unlikely]]: 
localhost:9001, queries: 10, QPS: 1.667, RPS: 166686.362, MiB/s: 1.272, result RPS: 0.000, result MiB/s: 0.000.

Without [[likely]] and [[unlikely]]: 
localhost:9001, queries: 10, QPS: 1.668, RPS: 166778.485, MiB/s: 1.272, result RPS: 0.000, result MiB/s: 0.000.
```